lib/symboldatabase.h

```diff
@@ -1153,6 +1153,7 @@ class CPPCHECKLIB ValueType {
     nonneg int bits;                    ///< bitfield bitcount
     nonneg int pointer;                 ///< 0=>not pointer, 1=>*, 2=>**, 3=>***, etc
     nonneg int constness;               ///< bit 0=data, bit 1=*, bit 2=**
+    bool reference = false;             ///< Is the outermost indirection of this type a reference
```

Comment:
We should distinguish between a lvalue and rvalue reference. This should probably be an enum.

Reply:
I agree. I've made this change and updated the parsing code to detect rvalue references. At this stage nothing uses RValue references but the option is there.